sparse_merkle_tree.js

```diff
@@ -71,14 +72,31 @@ class SparseMerkleTree {
         return new Promise((resolve, reject) => {
             in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (err, result) => {
                 if (err) {
-                    reject(err);
-                    return;
+                    return reject(err);
                 }
                 resolve(result);
             });
         });
     }
 
+    async verifyInclusionProof(root, queries, proof) {
+        for (let i = 0; i < queries.length; i++) {
+            if (!isInclusionProofForQueryKey(queries[i], proof.queries[i])) {
```

Comment:
Should we check that queries has the same length as proof.queries?

Reply:
It will be check in the verify function.

Reply:
Yes, but I would consider moving the check here (see LIP PR), I think it makes more sense here since we want to ensure that there is a correspondence between query keys and proof queries

Reply:
Almost all of our verifications are in Rust side so I think it is not a good option to move one of them in JS side.

Wdyt @shuse2 ?

Reply:
yea i agree
it's checked in the `verify` in rust [here](https://github.com/LiskHQ/lisk-db/blob/main/src/sparse_merkle_tree/smt.rs#L1547-L1549) along with other stuff, so since there is other stuff as well, i would keep the check there.